Add `pathFieldSeparator` support to `sortItemList`

velnic opened the pull request.

velnic committed:
Add separator to sortItemList

davidbonnet reviewed (changes requested):
Almost!

It should also update `queryItemList` to include the `separator` when calling `sortItemList`, and `QuerySettings` to include the `separatorEscape` string.

NicolasVelay commented:
Should the separator of `queryItemList` be in the query or an argument of `queryitemList`? I guess it should be in the query

davidbonnet commented:
> Should the separator of `queryItemList` be in the query or an argument of `queryitemList`? I guess it should be in the query

Yes, in the `query.settings`. Actually, `sortItemList` should have a similar signature to `testFilter`, where it takes a `settings` parameter instead `separator`, `separatorEscape`, etc.

velnic committed:
Use query.settings

velnic committed:
Reword prop type

davidbonnet committed:
Add `fieldSeparatorMap` setting

Remove `pathFieldSeparator`

davidbonnet committed:
Update documentation

davidbonnet committed:
Optimize `sortItemList`

davidbonnet committed:
Remove `fieldSeparatorMap`

Only use `pathField` (renamed from `pathFieldKey`),
`pathFieldSeparator`, and `pathFieldSeparatorEscape`.
